Common/src/interpolation_structure.cpp

```diff
@@ -1352,7 +1361,1770 @@ void CMirror::Set_TransferCoeff(CConfig **config) {
   }
 
   #ifdef HAVE_MPI
-  if (rank == MASTER_NODE) 
+  if (rank == MASTER_NODE && Buffer_Recv_mark != NULL) 
     delete [] Buffer_Recv_mark;
   #endif
 }
+
+CSlidingmesh::CSlidingmesh(CGeometry ***geometry_container, CConfig **config, unsigned int iZone, unsigned int jZone)  :  CInterpolator(geometry_container, config, iZone, jZone){
```

Comment:
Yep, you're right.
I created a couple of new methods, now the code should much more readable, let me know what you think.